src/mlpack/bindings/go/mlpack/arma_util.go

```diff
@@ -0,0 +1,396 @@
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaUmat(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  r, c := m.Dims()
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaUmat(C.CString(identifier), (*C.double)(ptr), C.int(c), C.int(r))
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaRow(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  e, err := m.Dims()
+  if err != 1 {
+  panic("Must have a single column")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaRow(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaUrow(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  e, err := m.Dims()
+  if err != 1 {
+  panic("Must have a single column")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaUrow(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaCol(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  err, e := m.Dims()
+  if err != 1 {
+  panic("Must have a single row")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaCol(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArma passes a gonum matrix to C by using it's gonums underlying blas64.
+func gonumToArmaUcol(identifier string, m *mat.Dense) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  err, e := m.Dims()
+  if err != 1 {
+  panic("Must have a single row")
+  }
+
+  blas64General := m.RawMatrix()
+  data := blas64General.Data
+
+  // Pass pointer of the underlying matrix to Mlpack.
+  ptr := unsafe.Pointer(&data[0])
+  C.mlpackToArmaUcol(C.CString(identifier), (*C.double)(ptr), C.int(e))
+}
+
+// GonumToArmaMatWithInfo passes a gonum matrix with info to C by 
+// using it's gonums underlying blas64.
+func gonumToArmaMatWithInfo(identifier string, m *DataWithInfo) {
+  // Get matrix dimension, underlying blas64General matrix, and data.
+  r, c := m.Data.Dims()
+  blas64General := m.Data.RawMatrix()
+  DataAndInfo := blas64General.Data
```

Comment:
This parameter is just the data, not the dimension info, so maybe `data` is a better name.  Also, should it start with a lowercase letter for consistency?

Reply:
Sorry It will not work. As after it start with a lower case it will become unexported.

Reply:
I'm a bit confused here; I thought in this line `DataAndInfo` referred to a local variable.  So in this case it wouldn't need to be exported?  I'm not much of a Go expert so please do correct me if I'm wrong...

Reply:
Sorry I talking about this `m.Data.RawMatrix()`.
I have updated `DataAndInfo` as suggested.